tools/nodeset_compiler/nodes.py

```diff
@@ -123,6 +125,17 @@ def parseXML(self, xmlelement):
                 if x.localName == "References":
                     self.parseXMLReferences(x)
 
+        # determine parent reference type
+        PARENT_REFERENCE_NODES = [
+                NodeId("ns=0;i=35"), NodeId("ns=0;i=45"),
+                NodeId("ns=0;i=46"), NodeId("ns=0;i=47"),
+                NodeId("ns=0;i=56"),
+        ]
+        for ref in self.references:
+            if ref.referenceType in PARENT_REFERENCE_NODES and not ref.isForward:
```

Comment:
The referenceType is of type str. Therefore the check will never be true. And what happens if more references match the condition?